Add C support

believeinlain opened the pull request:
PR that adds rudimentary support for C includes. Since C doesn't have a concept of "modules" we simply import items (which could be a function or variable) from files specified by path, with a boolean to distinguish system headers (such as `#include <stdio.h>`) from local headers (such as `#include "foo/bar.h"`).

Note that it considers `<stdio.h>` and `"stdio.h"` to be different include files. We can let the C compiler sort out correctness - naming local headers with the same names as system headers that one is also using is incredibly bad practice.

Added working example and doc test.

Addresses Issue #22

udoprog commented:
Looks great!

One thing I'm considering is whether the system header parameter might instead be better served by splitting it up into multiple function calls. But I'll have you decide since I don't use C codegen :).

believeinlain commented:
Ah, I didn't think of that! I think it would improve readability to have `include_header` and `include_system_header` rather than passing a boolean. I'll rework it.

udoprog commented:
Are things going OK? Would you like me to merge as-is?

believeinlain commented:
Sorry, I kinda got caught up in other stuff. I split `include` into `include` and `include_header` and updated the related docs and tests. It should be good to merge now.

udoprog commented on src/lang/c.rs:
```diff
@@ -0,0 +1,178 @@
+        ) -> fmt::Result {
+            let mut header = Tokens::new();
+
+            if let Some(package) = &config.package {
```
Just noticed, this looks accidentally carried over?

udoprog commented on src/lang/c.rs:
```diff
@@ -0,0 +1,178 @@
+/// Config data for C.
+#[derive(Debug, Default)]
+pub struct Config {
+    package: Option<ItemStr>,
```
And here.

udoprog commented:
Thank you!
